examples/finished/nodesel_hybridestim.py

```diff
@@ -0,0 +1,270 @@
+            (-1 for dynamic setting)
+        maxplungequot : float
+            Maximal quotient (curlowerbound - lowerbound)/(cutoffbound - lowerbound)
+            where plunging is performed
+        bestnodefreq : int
+            Frequency at which the best node instead of the hybrid best estimate/best bound
+            is selected (0: never)
+        estimweight : float
+            Weight of estimate value in node selection score
+            (0: pure best bound search, 1: pure best estimate search)
+        """
+        super().__init__()
+        self.scip = model
+        self.minplungedepth = minplungedepth
+        self.maxplungedepth = maxplungedepth
+        self.maxplungequot = maxplungequot
+        self.bestnodefreq = bestnodefreq if bestnodefreq > 0 else float('inf')
+        self.estimweight = estimweight
+
+    def _get_nodesel_score(self, node: Node) -> float:
+        """
+        Returns a weighted sum of the node's lower bound and estimate value.
+
+        Parameters
+        ----------
+        node : Node
+            The node to evaluate
+
+        Returns
+        -------
+        float
+            The node selection score
+        """
+        return ((1.0 - self.estimweight) * node.getLowerbound() +
+                self.estimweight * node.getEstimate())
+
+    def nodeselect(self):
+        """
+        Select the next node to process.
+
+        Returns
+        -------
+        dict
+            Dictionary with 'selnode' key containing the selected node
+        """
+        # Calculate minimal and maximal plunging depth
+        minplungedepth = self.minplungedepth
+        maxplungedepth = self.maxplungedepth
+
+        if minplungedepth == -1:
+            minplungedepth = self.scip.getMaxDepth() // 10
+            # Adjust based on strong branching iterations
+            if (self.scip.getNStrongbranchLPIterations() >
+                2 * self.scip.getNNodeLPIterations()):
+                minplungedepth += 10
+            if maxplungedepth >= 0:
+                minplungedepth = min(minplungedepth, maxplungedepth)
+
+        if maxplungedepth == -1:
+            maxplungedepth = self.scip.getMaxDepth() // 2
+
+        maxplungedepth = max(maxplungedepth, minplungedepth)
+
+        # Check if we exceeded the maximal plunging depth
+        plungedepth = self.scip.getPlungeDepth()
+
+        if plungedepth > maxplungedepth:
+            # We don't want to plunge again: select best node from the tree
+            if self.scip.getNNodes() % self.bestnodefreq == 0:
+                selnode = self.scip.getBestboundNode()
+            else:
+                selnode = self.scip.getBestNode()
+        else:
+            # Get global lower and cutoff bound
+            lowerbound = self.scip.getLowerbound()
+            cutoffbound = self.scip.getCutoffbound()
+
+            # If we didn't find a solution yet, use only 20% of the gap as cutoff bound
```

Comment:
```suggestion
            # If we didn't find a solution yet, tighten the cutoff bound to 20% of the range between it and the lowerbound.
```

It can be interpreted as only `0.2 * (cutoffbound - lowerbound)`, but I don't know how to improve it. but people can also just look at the freakin code